src/main/resources/mappings/forecast-state.json

```diff
@@ -0,0 +1,133 @@
+{
+  "dynamic": false,
+  "_meta": {
+    "schema_version": 1
+  },
+  "properties": {
+    "schema_version": {
+      "type": "integer"
+    },
+    "last_update_time": {
+      "type": "date",
+      "format": "strict_date_time||epoch_millis"
+    },
+    "error": {
+      "type": "text"
+    },
+    "started_by": {
+      "type": "keyword"
+    },
+    "stopped_by": {
+      "type": "keyword"
+    },
+    "forecaster_id": {
+      "type": "keyword"
+    },
+    "state": {
+      "type": "keyword"
+    },
+    "task_progress": {
+      "type": "float"
+    },
+    "init_progress": {
+      "type": "float"
+    },
+    "current_piece": {
+      "type": "date",
+      "format": "strict_date_time||epoch_millis"
+    },
+    "execution_start_time": {
+      "type": "date",
+      "format": "strict_date_time||epoch_millis"
+    },
+    "execution_end_time": {
+      "type": "date",
+      "format": "strict_date_time||epoch_millis"
+    },
+    "is_latest": {
+      "type": "boolean"
+    },
+    "task_type": {
+      "type": "keyword"
+    },
+    "checkpoint_id": {
+      "type": "keyword"
+    },
+    "coordinating_node": {
+      "type": "keyword"
+    },
+    "worker_node": {
+      "type": "keyword"
+    },
+    "user": {
+      "type": "nested",
+      "properties": {
+        "name": {
+          "type": "text",
+          "fields": {
+            "keyword": {
+              "type": "keyword",
+              "ignore_above": 256
+            }
+          }
+        },
+        "backend_roles": {
+          "type" : "text",
+          "fields" : {
+            "keyword" : {
+              "type" : "keyword"
+            }
+          }
+        },
+        "roles": {
+          "type" : "text",
+          "fields" : {
+            "keyword" : {
+              "type" : "keyword"
+            }
+          }
+        },
+        "custom_attribute_names": {
+          "type" : "text",
+          "fields" : {
+            "keyword" : {
+              "type" : "keyword"
+            }
+          }
+        }
+      }
+    },
+    "forecaster": {
+      FORECASTER_INDEX_MAPPING_PLACE_HOLDER
```

Comment:
separate PR?

Reply:
I have a function that replaces the string with concrete json contents:

    /**
     * Get forecaster state index mapping json content.
     *
     * @return forecaster state index mapping
     * @throws IOException IOException if mapping file can't be read correctly
     */
    public static String getStateMappings() throws IOException {
        String forecastStateMappings = getMappings(FORECAST_STATE_INDEX_MAPPING_FILE);
        String forecasterIndexMappings = getConfigMappings();
        forecasterIndexMappings = forecasterIndexMappings
            .substring(forecasterIndexMappings.indexOf("\"properties\""), forecasterIndexMappings.lastIndexOf("}"));
        return forecastStateMappings.replace("FORECASTER_INDEX_MAPPING_PLACE_HOLDER", forecasterIndexMappings);
    }

I did it similar to what AD does. The state mapping contains the mapping of config mapping. Since config mapping can be changing, it is better to use a placeholder string here and replace the actual contents dynamically.